### PR TITLE
Add / Edit template rules

### PR DESCRIPTION
**Pull request by mconnell**
Provides a way of adding and editing template rules.

### CODE REVIEW

**Comment by mconnell** [comment]
Updated to include UI work as the commit was quite small and didn't really need to be on a separate PR.

File: app/views/admin/rules/edit.html.erb [hunk 1]
```diff
@@ -9,7 +9,13 @@
 
 <h2>Source Rules</h2>
 
-<%= form_for @feed,
+<p>
+    Source rules cannot be edited locally. Please visit the
```
Comment (benbarnett) [inline app/views/admin/rules/edit.html.erb]: Indentation/line breaks a little weird.

File: app/views/admin/rules/edit.html.erb [hunk 2]
```diff
@@ -50,6 +56,39 @@
 
   <h2>Local Rules</h2>
```
Comment (benbarnett) [inline app/views/admin/rules/edit.html.erb]: Is this intentionally empty?

File: app/views/admin/rules/edit.html.erb [hunk 3]
```diff
@@ -50,6 +56,39 @@
...
+        <tr>
+          <th>Apply</th>
+          <th>Template Rule</th>
+        </tr>
+      </thead>
+      <tbody>
+        <% Rule.template_rules.each do |template_rule| %>
+          <tr>
+            <td>
+              <%= check_box_tag 'applied_rule_ids[]', template_rule.id, @feed.rule_applied?(template_rule) %>
```
Comment (benbarnett) [inline app/views/admin/rules/edit.html.erb]: These line breaks can cause weird gaps in older versions of IE. I would either keep them inline (as you have on lines 71/72) or consistently apply the block style to 71/72 like you have here.

Comment (benbarnett) [inline app/views/admin/rules/edit.html.erb]: (I know internal only – but as its quick)

Comment (mconnell) [inline app/views/admin/rules/edit.html.erb]: Thanks @benbarnett, fixing!

File: app/views/admin/rules/edit.html.erb [hunk 4]
```diff
@@ -29,19 +34,9 @@
...
-            <td>
-              <ul class="table__cell-list">
-                <% source_rule.sources.each do |source| %>
-                  <li><%= source.name %></li>
-                <% end %>
-              </ul>
-            </td>
+            <td><%= check_box_tag 'applied_rule_ids[]', source_rule.id, @feed.rule_applied?(source_rule) %></td>
+            <td><%= label_tag '', source_rule.name %></td>
+            <td><%= source_rule.sources.pluck(:name).join('<br>').html_safe %></td>
```
Comment (neoeno) [inline app/views/admin/rules/edit.html.erb]: Could be a helper, might make it more readable

File: app/views/admin/rules/edit.html.erb [hunk 5]
```diff
@@ -50,6 +45,34 @@
...
+          <th>Apply</th>
+          <th>Template Rule</th>
+        </tr>
+      </thead>
+      <tbody>
+        <% Rule.template_rules.each do |template_rule| %>
+          <tr>
+            <td>
+              <%= check_box_tag 'applied_rule_ids[]', template_rule.id, @feed.rule_applied?(template_rule) %></td>
+            <td><%= label_tag '', template_rule.name %></td>
```
Comment (neoeno) [inline app/views/admin/rules/edit.html.erb]: There's a label tag without a target here (and a few other places). Is there a reason it's not targeted?

File: spec/controllers/admin/template_rules_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+    context 'failed update' do
+      let(:update_result) { false }
+
+      it 're-renders the edit page' do
+        expect(response).to render_template(:edit)
+      end
+    end
+  end
+end
```
Comment (neoeno) [inline spec/controllers/admin/template_rules_controller_spec.rb]: nice clean specs. a pleasure!

File: spec/factories/rule_factories.rb [hunk 7]
```diff
@@ -9,5 +9,9 @@
       source_rule true
       sources { |o| [o.association(:source)] }
     end
+
+    trait :template_rule do
+      template_rule true
+    end
```
Comment (neoeno) [inline spec/factories/rule_factories.rb]: I learned something today.

File: app/views/admin/rules/edit.html.erb (hunk 4 above)
Comment (urfolomeus) [inline app/views/admin/rules/edit.html.erb]: Already did the refactor this afternoon but figured it was too much for one PR. :)

File: app/views/admin/rules/edit.html.erb (hunk 5 above)
Comment (urfolomeus) [inline app/views/admin/rules/edit.html.erb]: Hmm, good point. Not sure how we missed that. Thanks for catching >.<

**Comment by mconnell** [comment]
@neoeno fixed the labels not correctly being associated with the checkboxes

**Comment by munckymagik** [comment]
:+1:

**Comment by neoeno** [comment]
I guess checkbox_tag doesn't give you a default ID like the methods on form then? Shame.

:+1:

**Comment by benbarnett** [comment]
:+1:

**Comment by urfolomeus** [comment]
Had to rebase the branch after previous PR was merged. No changes otherwise.

**Pull request merged**